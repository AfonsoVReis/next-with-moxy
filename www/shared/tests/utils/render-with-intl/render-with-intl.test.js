import React from 'react';
import { FormattedMessage } from 'react-intl';

import { renderWithIntl } from './render-with-intl';

it('should correctly render formatted message', () => {
    const { container, getByText } = renderWithIntl(
        <FormattedMessage id={ 'foo' } />,
    );

    expect(container.innerHTML).toMatchSnapshot();
    expect(getByText('foo')).toBeInTheDocument();
});

it('should correctly rerender with passed children', () => {
    const { rerenderWithIntl, getByText } = renderWithIntl(
        <FormattedMessage id={ 'foo' } />,
    );

    rerenderWithIntl('bar');

    expect(getByText('bar')).toBeInTheDocument();
});
