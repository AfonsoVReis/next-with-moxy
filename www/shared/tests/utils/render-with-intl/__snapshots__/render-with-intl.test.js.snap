// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render with Intl should correctly render formatted message 1`] = `"foo"`;
